File: src/main/java/fr/uiytt/ageofempire/ConfigManager.java
package fr.uiytt.ageofempire;

import org.bukkit.Location;
import org.bukkit.Material;
import org.bukkit.Tag;
import org.bukkit.World;

import java.util.HashSet;
import java.util.List;
import java.util.Set;

/**
 * Handle the main config file
 * Currently not loading the config, need to be improved
 */
public class ConfigManager {

    //private static final Yaml configYaml = new Yaml("config.yml", "plugins" + File.separator + "AgeOfEmpire");
    private static World world;
    private static int pvpTimer = 20*60;
    private static int assaultTimer = 40*60;

    private static final Set<Material> breakableBlocks = new HashSet<>(List.of(Material.ANDESITE));
    private static final Set<Material> deletedDrops = new HashSet<>(List.of(Material.DIRT));
    private static final Set<Location> seaLanterns = new HashSet<>();

    public static void init() {
        breakableBlocks.addAll(Tag.OAK_LOGS.getValues());
        breakableBlocks.addAll(Tag.PLANKS.getValues());
        breakableBlocks.add(Material.SEA_LANTERN);
        breakableBlocks.add(Material.CAKE);
        deletedDrops.addAll(Tag.WOOL.getValues());
        deletedDrops.addAll(Tag.ITEMS_STONE_TOOL_MATERIALS.getValues());
        deletedDrops.add(Material.LEATHER_HELMET);
        deletedDrops.add(Material.LEATHER_CHESTPLATE);
        deletedDrops.add(Material.LEATHER_LEGGINGS);
        deletedDrops.add(Material.LEATHER_BOOTS);

        seaLanterns.add(new Location(getWorld(), 0, 72, 0));
    }

    public static World getWorld() {
        return world;
    }
    public static void setWorld(World world) {
        ConfigManager.world = world;
    }

    public static int getPvpTimer() {
        return pvpTimer;
    }
    public static void setPvpTimer(int pvpTimer) {
        ConfigManager.pvpTimer = pvpTimer;
    }

    public static int getAssaultTimer() {
        return assaultTimer;
    }
    public static void setAssaultTimer(int assaultTimer) {
        ConfigManager.assaultTimer = assaultTimer;
    }

    public static Set<Material> getBreakableBlocks() {
        return breakableBlocks;
    }

    public static Set<Material> getDeletedDrops() { return deletedDrops; }
    public static Set<Location> getSeaLanterns() {return seaLanterns;}
}

File: src/main/kotlin/fr/uiytt/ageofempire/ConfigManager.kt
package fr.uiytt.ageofempire

import org.bukkit.*

/**
 * Handle the main config file
 * Currently not loading the config, need to be improved
 */
class ConfigManager {
    //private static final Yaml configYaml = new Yaml("config.yml", "plugins" + File.separator + "AgeOfEmpire") 
    var world: World? = Bukkit.getWorld("world")
    var pvpTimer = 20 * 60
    var assaultTimer = 40 * 60

    private val breakableBlocks: MutableSet<Material> = HashSet(listOf(Material.ANDESITE))
    private val deletedDrops: MutableSet<Material> = HashSet(listOf(Material.DIRT))
    private val seaLanterns: MutableSet<Location> = HashSet() //Sea lanterns are infinite ressources

    fun init() {
        breakableBlocks.addAll(Tag.OAK_LOGS.values)
        breakableBlocks.addAll(Tag.PLANKS.values)
        deletedDrops.addAll(Tag.WOOL.values)
        deletedDrops.addAll(Tag.ITEMS_STONE_TOOL_MATERIALS.values)
        breakableBlocks.add(Material.SEA_LANTERN)
        breakableBlocks.add(Material.CAKE)
        deletedDrops.add(Material.LEATHER_HELMET)
        deletedDrops.add(Material.LEATHER_CHESTPLATE)
        deletedDrops.add(Material.LEATHER_LEGGINGS)
        deletedDrops.add(Material.LEATHER_BOOTS)
        seaLanterns.add(Location(world, 0.0, 72.0, 0.0))
    }

    fun getSetOfBreakableBlocks(): Set<Material> = breakableBlocks.toSet() 
    fun getSetOfDeletedDrops(): Set<Material> = deletedDrops.toSet() 
    fun getSetOfSeaLanterns(): Set<Location> = seaLanterns.toSet() 

}
fun getConfigManager(): ConfigManager = AgeOfEmpire.configManager

/**
 * For a given string of coordinates, return a location in the Config World
 * @param locString a string either "x y z" or "x y z yaw pitch"
 * @return a location in [ConfigManager.world]
 */
fun stringToLocation(locString: String): Location {
    val string = locString.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
    return try {
        val loc = Location(getConfigManager().world, string[0].toDouble(), string[1].toDouble(), string[2].toDouble())
        if (string.size > 3) {
            loc.yaw = string[3].toFloat()
            loc.pitch = string[4].toFloat()
        }
        loc
    } catch (exception: NumberFormatException) {
        exception.printStackTrace()
        Location(getConfigManager().world, 0.0, 80.0, 0.0)
    }
}
